### ISSUE TITLE
zfs_vdev_raidz_impl module parameter rejects avx2 on module load

### ISSUE THREAD

**Issue by ofaaland** [zfsonlinux/zfs#4753]
I see in the manual page zfs-module-parameters.5 states that, avx2 is not an allowed value for module parameter zfs_vdev_raidz_impl.  However, if the user selects "fastest", zfs may choose to use avx2 (and it does on my test system).

Were the module init code and the manpage were left this way accidentally?

**Comment by DeHackEd** [comment]
AVX[2] for RAID-Z has not been merged. Do you mean fletcher4?

Instruction support is dependent on compiler and assembler support as well. Make sure your compiler and assembler support AVX2.

**Comment by ofaaland** [comment]
Nope, I do mean zfs_vdev_raidz_impl; I pulled it into a local branch for testing.  Thanks though.

It works fine so far - I set the module parameter to "fastest" when loading the module or set it to "avx2" after loading, and then can see with perf top:
{noformat}
   2.25%  [kernel]                   [k] avx2_gen_pq
{noformat}

And furthermore can see that the %cpu taken is much less with the avx2 version than with the original version.

I'm just talking about the code for handling zfs_vdev_raidz_impl=avx2 when the module is loaded.
